packages/pg/lib/client.js

```diff
@@ -479,9 +486,22 @@ class Client extends EventEmitter {
   }
 
   createConnectionMap(data) {
-    Client.connectionMap.clear()
+    let currConnectionMap = Client.connectionMap
```

Comment:
Should we still clear the `connectionMap` after copying it into `currConnectionMap`? There can be a server which has been removed from the cluster but the driver has not yet marked it as `failedHost`. Such a host would be in `connectionMap` but not in `data` or `failedHosts`.
Eventually, it'll be marked as failed anyways when it'll be attempted even if you do not clear it here.